vspace/src/main/java/edu/asu/diging/vspace/core/services/impl/SlideManager.java

```diff
@@ -46,4 +53,25 @@ public ISlide getSlide(String slideId) {
     public void updateSlide(Slide slide) {
         slideRepo.save((Slide) slide);
     }
+
+    @Override
+    public void deleteSlideById(String slideId, String moduleId) throws SlideDoesNotExistException {
+
+        try {
+            List<Sequence> sequences = sequenceRepo.findSequencesForModule(moduleId);
+            outer: for(Sequence sequence : sequences) {
+                for(ISlide slideIds : sequence.getSlides()) {
+                    if(slideIds.getId().equals(slideId)) {
+                        sequence.getSlides().remove(slideIds);
+                        sequenceRepo.save(sequence);
+                        continue outer;
```

Comment:
I wouldn't do that. Right now it's not possible to have a slide appear more than once in a sequence, but it might make sense to change that.